packages/utils/src/jwk-did-utils.ts

```diff
@@ -112,3 +113,112 @@ export function generateJwkFromVerificationMethod(
   const { publicKeyHex, keyType: extractedType } = extractPublicKeyHex(key)
   return createJWK(keyType, publicKeyHex, keyUse)
 }
+
+const isString = (value: any): value is string => typeof value === 'string'
+
+/**
+ * Validate the JOSE header of a JWT
+ * Only support kid with DID URL
+ * @param header
+ * @param payload
+ * Ref: https://www.w3.org/TR/vc-jose-cose/#using-header-params-claims-key-discovery
+ * Ref: https://www.w3.org/TR/vc-jose-cose/#jose-header-parameters-jwt-claims
+ */
+export const validateHeaderJOSE = (header: any, payload: any) => {
+  // Check for 'alg' (Algorithm - Required)
+  if (!isString(header.alg)) {
+    throw new Error('alg must be present and a string')
+  }
+
+  // Check for 'kid' (Key ID - Required)
+  // https://www.w3.org/TR/vc-jose-cose/#kid
+  if (!isString(header.kid)) {
+    throw new Error('kid must be a string')
+  }
+  const isDidUrlKey = /^did:\w+:/.test(header.kid)
+  if (!isDidUrlKey && header.kid.startsWith('did:')) {
+    throw new Error('kid must be a full DID URL when starting with "did:"')
+  }
+
+  // Check for 'iss' (Issuer - Optional)
+  // https://www.w3.org/TR/vc-jose-cose/#iss
+  if ('iss' in header) {
+    if (!('issuer' in payload)) {
+      throw new Error('issuer must be present in payload when iss is in header')
+    }
+
+    let issuerId = extractIssuer(payload, { removeParameters: true })
+    // Check if 'iss' matches 'issuer.id' or 'issuer' string
+    if (issuerId && header['iss'] !== issuerId) {
+      throw new Error('iss in header does not match issuer value')
+    }
+  }
+
+  // Check for 'cty' (Content Type - Optional)
+  // https://www.w3.org/TR/vc-jose-cose/#securing-with-jose
+  if ('cty' in header) {
+    if (!isString(header.cty)) {
+      throw new Error('cty must be a string')
+    }
+
+    if (header.cty !== 'vc') {
+      throw new Error('cty must be "vc"')
+    }
+  }
+
+  // Check for 'typ' (Type - Optional)
+  // https://www.w3.org/TR/vc-jose-cose/#securing-with-jose
+  if ('typ' in header) {
+    if (!isString(header.typ)) {
+      throw new Error('typ must be a string')
+    }
+
+    if (header.typ !== 'vc+jwt') {
+      throw new Error('typ must be "vc+jwt"')
+    }
```

Comment:
Based on this spec: https://www.w3.org/TR/vc-jose-cose/#securing-with-jose,
I’m not sure if the `typ` must be the string `vc+jwt`, since the spec says it SHOULD BE. Could you check it again to ensure this if statement is correct?